pkg/providers/vsphere/internal/templates/factory.go

```diff
@@ -148,3 +167,143 @@ func (f *Factory) importOVAIfMissing(ctx context.Context, templateName, ovaURL s
 
 	return nil
 }
+
+func (f *Factory) resizeDisk(ctx context.Context, templateFullPath, templateName, datacenter string) error {
+
+	// Get devices information template to identify second disk properly
+	logger.V(2).Info("Getting devices info for template")
+	devicesInfo, err := f.client.DevicesInfo(ctx, datacenter, templateName)
+
+	if err != nil {
+		return err
+	}
+	// For 1.22 we switched to using one disk for BR, so it for now as long as the boolean is set, and we only see
+	// one disk, we can assume this is for 1.22. This loop would need to change if that assumption changes
+	// in the future, but 1.20 and 1.21 are still using dual disks which is why we need to check for the second
+	// disk first. Since this loop will get all kinds of devices and not just hard disks, we need to do these
+	// checks based on the label.
+	disk1 := ""
+	disk2 := ""
+	alreadySized := false
+	for _, deviceInfo := range devicesInfo.([]interface{}) {
+		deviceMetadata := deviceInfo.(map[string]interface{})["DeviceInfo"]
+		deviceLabel := deviceMetadata.(map[string]interface{})["Label"].(string)
+		deviceDiskSize := deviceMetadata.(map[string]interface{})["Summary"].(string)
+
+		// Get the name of the hard disk and resize the disk to 20G
+		if strings.EqualFold(deviceLabel, "Hard disk 1") {
+			disk1 = deviceInfo.(map[string]interface{})["Name"].(string)
+			parsedDiskSize := parseDiskSize(deviceDiskSize)
+
+			if parsedDiskSize == -1 {
+				logger.V(2).Info("Unable to get converted KB to GB int")
+				return fmt.Errorf("unable to get converted KB to GB int: %d", parsedDiskSize)
+			}
+
+			if parsedDiskSize == 23 {
+				alreadySized = true
+			}
+		} else if strings.EqualFold(deviceLabel, "Hard disk 2") {
+			disk2 = deviceInfo.(map[string]interface{})["Name"].(string)
+			parsedDiskSize := parseDiskSize(deviceDiskSize)
+
+			if parsedDiskSize == -1 {
+				logger.V(2).Info("Unable to get converted KB to GB int")
+				return fmt.Errorf("unable to get converted KB to GB int: %d", parsedDiskSize)
+			}
+
+			if parsedDiskSize == 20 {
+				alreadySized = true
+			}
+			break
+		}
+	}
+
+	if alreadySized {
+		return nil
+	}
+	diskName := ""
+	var diskSizeInGB int
+	if disk2 != "" {
+		logger.V(2).Info("Resizing disk 2 of template to 20G")
+		diskName = disk2
+		diskSizeInGB = 20
+	} else if disk1 != "" {
+		logger.V(2).Info("Resizing disk 1 of template to 22G")
+		diskName = disk1
+		diskSizeInGB = 22
+	} else {
+		return fmt.Errorf("template %v is not valid as there are no associated disks", templateName)
+	}
```

Comment:
all this logic seems very similar to the one in govc.go
Assuming this is necessary, could we get a small writeup in the PR description about what all of this is doing and why is necessary? It might help understand the implementation :)